heron/schedulers/src/java/com/twitter/heron/scheduler/ecs/EcsScheduler.java

```diff
@@ -0,0 +1,183 @@
+
+  }
+
+  public void close() {
+
+  }
+
+
+  protected int startExecutorSyncProcess(int container) {
+    String executingInShell = new String();
+    executingInShell = getExecutorCommand(container)[0];
+    System.out.println("executing in Shell: " + executingInShell);
+    return ShellUtils.runProcess(executingInShell, null);
+  }
+
+  protected void startExecutor(final int container) {
+    LOG.info("Starting a new executor for container: " + container);
+    int shellOutput = startExecutorSyncProcess(container);
+    LOG.info("output value for the executor container: "
+        + container + String.valueOf(shellOutput));
+
+  }
+
+  private String[] getExecutorCommand(int container) {
+    List<Integer> freePorts = new ArrayList<>(SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR);
+    for (int i = 0; i < SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR; i++) {
+      //freePorts.add(SysUtils.getFreePort());
+      freePorts.add(5000 + (i + (container * 10)));
+    }
+    String[] executorCmd = SchedulerUtils.executorCommand(config, runtime, container, freePorts);
+    String finalExecCommand = setClusterValues(formHeronExecCommand(executorCmd));
+    String ecsTaskProject =  EcsContext.topologyName(config) + "_" + String.valueOf(container);
+    FileOutputStream dockerFilestream = null;
+    String content = null;
+    try {
+      tempDockerFile = File.createTempFile("docker", ".yml");
+      content = getDockerFileContent(finalExecCommand, container);
+      tempDockerFile.setWritable(true);
+      dockerFilestream = new FileOutputStream(tempDockerFile);
+      IOUtils.write(content, dockerFilestream);
+      IOUtils.closeQuietly(dockerFilestream);
+    } catch (IOException  e) {
+      e.printStackTrace();
+    } finally {
+      IOUtils.closeQuietly(dockerFilestream);
+    }
+    String dockerComposeFileName = " --file " + tempDockerFile;
+    String finalCommand = EcsContext.COMPOSE_CMD + ecsTaskProject + dockerComposeFileName;
+    finalCommand = finalCommand + EcsContext.UP;
+    System.out.println("final Ecs Task command " + finalCommand);
+    tempDockerFile.deleteOnExit();
+    return  new String[] {finalCommand};
+
+  }
+
+  public String setClusterValues(String localExecCommand) {
+    String clusterExecCommand = localExecCommand.replace(Context.topologyBinaryFile(config),
+                                                           EcsContext.ECS_CLUSTER_BINARY);
+    // line below can be removed once the Cluster JVM TODO is resolved
+    clusterExecCommand = clusterExecCommand.replace(Context.clusterJavaHome(config),
+                                                     EcsContext.DESTINATION_JVM);
+    clusterExecCommand = clusterExecCommand.replaceAll("\"", "'");
+    return clusterExecCommand;
+  }
+
+  public String replacePortNumbers(int container, String content) {
+    int basePortnumber = 5000;
+    String localContent = new String(content);
+    for (int i = 0; i < SchedulerUtils.PORTS_REQUIRED_FOR_EXECUTOR; i++) {
```

Comment:
please clarify in the javadocs what this method does. Also I suspect StringBuilder might be a better fit here.